File: seoyeon/week5/영어 끝말잇기.py
def solution(n, words):
    answer = [0, 0]
    tmp = []
    use = []
    for i in range(0, len(words)-1) :
        tmp.append(words[i])
        tmp.append(words[i+1])
        if (tmp[0][-1] != tmp[1][0]) or tmp[1] in use :
            answer = [(i+1)%n+1, (i+1)//n+1]
            break
        use.append(tmp[0])
        tmp.clear()
    return answer

File: hyeyoung/week5/영어끝말잇기.java
package hyeyoung.week5;

import java.util.HashMap;
import java.util.Map;

public class 영어끝말잇기 {
}

class Solution2 {
    public int[] solution(int n, String[] words) {
        int[] answer = new int[2];

        Map<String, Integer> map = new HashMap<>();

        for(int i = 0; i < words.length; i++){
            if(i != 0){
                String s1 = words[i - 1];
                String s2 = words[i];

                if(map.containsKey(s2) || s1.charAt(s1.length() - 1) != s2.charAt(0)){
                    answer[0] = (i % n) + 1;
                    answer[1] = (i / n) + 1;

                    return answer;
                }
            }
            map.put(words[i], 1);
        }

        return answer;
    }
}